app/scripts/utils/antimeridian.ts

```diff
@@ -0,0 +1,213 @@
+    }
+  } as Feature<MultiPolygon>;
+}
+
+/**
+ * Returns the rings of a MultiPolygon that are before -180 if direction is
+ * 'west', and after 180 if direction is 'east'.
+ * @param feature - The feature to check.
+ * @param direction - The direction to check for (either 'west' or 'east').
+ * @returns The MultiPolygon with the correct rings, or null if there are no
+ * rings left after the filtering
+ */
+function getOutsidePolygons(
+  feature: Feature<MultiPolygon> | null,
+  direction: Direction
+) {
+  if (!feature) return null;
+
+  const checkRing = (ring: Position[]): boolean =>
+    ring.some(([lng]) => {
+      if (direction === 'west') {
+        return lng < -180;
+      } else {
+        return lng > 180;
+      }
+    });
+
+  const outsidePolygons = feature.geometry.coordinates.filter(
+    (poly: Position[][]) => poly.some(checkRing)
+  );
+
+  return outsidePolygons.length
+    ? ({
+        type: 'Feature',
+        properties: {},
+        geometry: {
+          type: 'MultiPolygon',
+          coordinates: outsidePolygons
+        }
+      } as Feature<MultiPolygon>)
+    : null;
+}
+
+/**
+ * Shifts a ring of coordinates either east or west.
+ * This shift is not automatic. The caller needs to know which direction to shift.
+ * For rings east of the 180, shift the coordinates west. The ring will be
+ * shifted until the smallest lng is below 180.
+ * For rings west of the -180, shift the coordinates east. The ring will be
+ * shifted until the largest lng is above -180.
+ *
+ * @param ring - The ring of coordinates to shift.
+ * @param direction - The direction to shift the ring. Either 'east' or 'west'.
+ * @returns A new ring of coordinates shifted
+ */
+function shiftRing(ring: Position[], direction: Direction): Position[] {
+  if (direction === 'west') {
+    // We need to shift the ring west, but we want to shift the whole ring.
+    // So we need to find out by how many times we need to shift 360. This will
+    // be defined by the smallest lng value. Any coordinates that are still
+    // outside after the shift will be shifted again in a later run.
+    const smallestLng = Math.min.apply(
+      null,
+      ring.map(([lng]) => lng)
+    );
+
+    // Or 1 in case the smallestLng is exactly 180.
+    const shift = Math.ceil(Math.abs((smallestLng - 180) / 360)) || 1;
+
+    return ring.map(([lng, lat]) => {
+      return [lng - shift * 360, lat];
+    });
+  } else {
+    // We need to shift the ring east, but we want to shift the whole ring.
+    // So we need to find out by how many times we need to shift 360. This will
+    // be defined by the smallest lng value. Any coordinates that are still
+    // outside after the shift will be shifted again in a later run.
+    const largestLng = Math.max.apply(
+      null,
+      ring.map(([lng]) => lng)
+    );
+
+    // Or 1 in case the largestLng is exactly -180.
+    const shift = Math.ceil(Math.abs((largestLng + 180) / 360)) || 1;
```

Comment:
Is it okay to round up like this? 1 degree in Mercator doesn't seem to be that small number?

Reply:
This does not refer to 1º but to how many times to ship 360º. The result is that the coordinates move to the "repeated map" they are to the correct one (between -180/180)